### PR TITLE
[arp_update]: Set failed IPv6 neighbors to incomplete

### PR DESCRIPTION
**Pull request by theasianpianist**
Signed-off-by: Lawrence Lee <lawlee@microsoft.com>

#### Why I did it

If IPv6 neighbor entries become "FAILED", then any subsequent neighbor advertisement messages will not be able to resolve these entries in the neighbor cache, only a ping or other traffic being sent from the device will be able to resolve these. This means that standby ToRs may have such FAILED entries stuck in the neighbor cache for a long time.

#### How I did it

After pinging any failed IPv6 neighbor entries, set the remaining failed/incomplete entries to a permanent INCOMPLETE state. This manual setting to INCOMPLETE prevents these entries from automatically transitioning to FAILED state, and since they are now incomplete any subsequent NA messages for these neighbors is able to resolve the entry in the cache.

#### How to verify it

1. Add a failed IPv6 neighbor entry to the neighbor cache (e.g. pinging an unreachable address)
2. Run the arp_update script
3. The entry should now be incomplete

```
root@str2-7050cx3-acs-08:/# ping fc02:1000::500
PING fc02:1000::500(fc02:1000::500) 56 data bytes
^C
--- fc02:1000::500 ping statistics ---
2 packets transmitted, 0 received, 100% packet loss, time 13ms

root@str2-7050cx3-acs-08:/# ip neigh show fc02:1000::500
fc02:1000::500 dev Vlan1000  FAILED
root@str2-7050cx3-acs-08:/# arp_update
ping6: Warning: source address might be selected on device other than PortChannel101.
ping6: Warning: source address might be selected on device other than PortChannel102.
ping6: Warning: source address might be selected on device other than PortChannel103.
ping6: Warning: source address might be selected on device other than PortChannel104.
ping6: Warning: source address might be selected on device other than Vlan1000.
^C
root@str2-7050cx3-acs-08:/# ip neigh show fc02:1000::500
fc02:1000::500 dev Vlan1000  INCOMPLETE
root@str2-7050cx3-acs-08:/# sleep 10
root@str2-7050cx3-acs-08:/# ip neigh show fc02:1000::500
fc02:1000::500 dev Vlan1000  INCOMPLETE
root@str2-7050cx3-acs-08:/#
```

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [x] 202012
- [ ] 202106
- [ ] 202111
- [x] 202205

### CODE REVIEW

File: files/scripts/arp_update [hunk 1]
```diff
@@ -93,6 +93,18 @@ while /bin/true; do
...
+        # manually set any remaining FAILED/INCOMPLETE entries to permanently INCOMPLETE
+        # this prevents any remaining INCOMPLETE entries from automatically transitioning to FAILED
+        # once these entries are incomplete, any subsequent neighbor advertisement messages
+        # are able to resolve the entry
+
+        # generates the following command for each failed or incomplete IPv6 neighbor
+        # ip neigh replace <neighbor IPv6> dev <VLAN name> nud incomplete
+        neigh_replace_template="sed -e 's/^/ip neigh replace /' -e 's/,/ dev /' -e 's/$/ nud incomplete;/'" 
+        ip_neigh_replace_cmd="ip -6 neigh show | grep -v fe80 | grep Vlan1000 | grep -E 'FAILED|INCOMPLETE' | cut -d ' ' -f 1,3 --output-delimiter=',' | $neigh_replace_template"
+        eval `eval $ip_neigh_replace_cmd`
```
Comment (prsunny) [inline files/scripts/arp_update:106]: Let's do this prior to sending ping (line 95). We don't want to end up in a situation where ping succeeded, then set to INCOMPLETE by the script due to some timing issue. So lets first set to INCOMPLETE and then send ping which is safe.

**Review by prsunny — APPROVED** [review]

**Comment by prsunny** [comment]
Can you please raise one for 202012?

**Comment by theasianpianist** [comment]
> Can you please raise one for 202012?

Addressed offline, we should be able to cherry-pick to 202012 without any conflicts

**Pull request merged**